python/pyproject.toml

```diff
@@ -1,20 +1,46 @@
 [build-system]
-requires = ["maturin>=1.3.0,<1.4.0", "setuptools", "wheel", "Cython"]
+requires = ["maturin>=1.5.1", "setuptools", "wheel", "Cython"]
 build-backend = "maturin"
 
 [project]
 name = "bystro"
-requires-python = ">=3.10"
+requires-python = ">=3.11.0"
```

Comment:
We could lower this back to 3.10, but I would argue we shouldn't unless there is a user need: python 3.11 is notably faster than python 3.10, introduces some useful syntax, and is now a year old. Additionally all of our CI is against 3.11, so we haven't actually been testing 3.10.